exporters/otlp/trace/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpGrpcSpanExporterBuilder.java

```diff
@@ -114,6 +129,51 @@ public OtlpGrpcSpanExporter build() {
         managedChannelBuilder.intercept(MetadataUtils.newAttachHeadersInterceptor(metadata));
       }
 
+      if (trustedCertificatesPem != null) {
+        // gRPC does not abstract TLS configuration so we need to check the implementation and act
+        // accordingly.
+        if (managedChannelBuilder
+            .getClass()
+            .getName()
+            .equals("io.grpc.netty.NettyChannelBuilder")) {
+          NettyChannelBuilder nettyBuilder = (NettyChannelBuilder) managedChannelBuilder;
+          try {
+            nettyBuilder.sslContext(
+                GrpcSslContexts.forClient()
+                    .trustManager(new ByteArrayInputStream(trustedCertificatesPem))
+                    .build());
+          } catch (IllegalArgumentException | SSLException e) {
+            throw new IllegalStateException(
+                "Could not set trusted certificates for gRPC TLS connection, are they valid "
+                    + "X.509 in PEM format?",
+                e);
+          }
+        } else if (managedChannelBuilder
+            .getClass()
+            .getName()
+            .equals("io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder")) {
+          io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder nettyBuilder =
+              (io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder) managedChannelBuilder;
+          try {
+            nettyBuilder.sslContext(
+                io.grpc.netty.shaded.io.grpc.netty.GrpcSslContexts.forClient()
+                    .trustManager(new ByteArrayInputStream(trustedCertificatesPem))
+                    .build());
+          } catch (IllegalArgumentException | SSLException e) {
+            throw new IllegalStateException(
+                "Could not set trusted certificates for gRPC TLS connection, are they valid "
+                    + "X.509 in PEM format?",
+                e);
+          }
+        } else {
+          throw new IllegalStateException(
+              "TLS cerificate configuration only supported with Netty. "
+                  + "If you need to configure a certificate, switch to grpc-netty or "
+                  + "grpc-netty-shaded.");
+        }
+        // TODO(anuraaga): Support okhttp.
```

Comment:
can you add an issue for this, so we don't forget?

Reply:
Sure filed https://github.com/open-telemetry/opentelemetry-java/issues/2430